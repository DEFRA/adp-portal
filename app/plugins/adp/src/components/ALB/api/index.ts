export { ArmsLengthBodyClient } from './AlbClient'
export type {ArmsLengthBodyApi} from './AlbApi'
