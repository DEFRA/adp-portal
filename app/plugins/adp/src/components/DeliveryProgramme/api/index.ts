export { DeliveryProgrammeClient } from './DeliveryProgrammeClient';
export type { DeliveryProgrammeApi } from './DeliveryProgrammeApi';
