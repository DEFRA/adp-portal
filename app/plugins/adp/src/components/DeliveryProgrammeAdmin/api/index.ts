export { deliveryProgrammeAdminApiRef } from './DeliveryProgrammeAdminApi';
export { DeliveryProgrammeAdminApiClient } from './DeliveryProgrammeAdminClient';
