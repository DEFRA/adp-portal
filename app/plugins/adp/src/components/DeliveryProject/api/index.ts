export * from './DeliveryProjectClient';
export * from './DeliveryProjectApi';
