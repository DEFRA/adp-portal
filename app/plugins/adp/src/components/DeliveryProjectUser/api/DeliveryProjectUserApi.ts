import type {
  CreateDeliveryProjectUserRequest,
  DeliveryProjectUser,
} from '@internal/plugin-adp-common';
import { createApiRef } from '@backstage/core-plugin-api';

export const deliveryProjectUserApiRef = createApiRef<DeliveryProjectUserApi>({
  id: 'plugin.adp.deliveryprojectuserapi',
});

export interface DeliveryProjectUserApi {
  getAll(): Promise<DeliveryProjectUser[]>;
  getByDeliveryProjectId(
    deliveryProjectId: string,
  ): Promise<DeliveryProjectUser[]>;
  create(data: CreateDeliveryProjectUserRequest): Promise<DeliveryProjectUser>;
}
