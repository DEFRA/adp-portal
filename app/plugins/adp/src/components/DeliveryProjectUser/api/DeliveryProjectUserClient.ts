import type {
  DeliveryProjectUser,
  CreateDeliveryProjectUserRequest,
  UpdateDeliveryProjectUserRequest,
  DeleteDeliveryProjectUserRequest,
} from '@internal/plugin-adp-common';
import type { DeliveryProjectUserApi } from './DeliveryProjectUserApi';
import type { DiscoveryApi, FetchApi } from '@backstage/core-plugin-api';
import { ResponseError } from '@backstage/errors';
import { ValidationError } from '../../../utils/ValidationError';

export class DeliveryProjectUserClient implements DeliveryProjectUserApi {
  readonly #discoveryApi: DiscoveryApi;
  readonly #fetchApi: FetchApi;

  constructor(discoveryApi: DiscoveryApi, fetchApi: FetchApi) {
    this.#discoveryApi = discoveryApi;
    this.#fetchApi = fetchApi;
  }

  async getAll(): Promise<DeliveryProjectUser[]> {
    const url = await this.#getBaseUrl();

    const response = await this.#fetchApi.fetch(url);

    if (!response.ok) {
      throw await ResponseError.fromResponse(response);
    }

    const deliveryProjectUsers = asProjectUsers(await response.json());

    for (const item of deliveryProjectUsers)
      item.updated_at = new Date(item.updated_at);

    return deliveryProjectUsers;
  }

  async getByDeliveryProjectId(
    deliveryProjectId: string,
  ): Promise<DeliveryProjectUser[]> {
    const baseUrl = await this.#getBaseUrl();
    const url = `${baseUrl}/${deliveryProjectId}`;
    const response = await this.#fetchApi.fetch(url);

    if (!response.ok) {
      throw await ResponseError.fromResponse(response);
    }

    const deliveryProgrammeAdmins = asProjectUsers(await response.json());

    return deliveryProgrammeAdmins;
  }

  async create(
    data: CreateDeliveryProjectUserRequest,
  ): Promise<DeliveryProjectUser> {
    const url = await this.#getBaseUrl();

    const response = await this.#fetchApi.fetch(url, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(data),
    });

    return await this.#handleCreateUpdateResponse(response);
  }

  async update(
    data: UpdateDeliveryProjectUserRequest,
  ): Promise<DeliveryProjectUser> {
    const url = await this.#getBaseUrl();

    const response = await this.#fetchApi.fetch(url, {
      method: 'PATCH',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(data),
    });

    return await this.#handleCreateUpdateResponse(response);
  }

  async #getBaseUrl(): Promise<string> {
    return `${await this.#discoveryApi.getBaseUrl('adp')}/deliveryProjectUsers`;
  }

  async #handleCreateUpdateResponse(
    response: Response,
  ): Promise<DeliveryProjectUser> {
    if (response.status === 400)
      throw new ValidationError((await response.json()).errors);

    if (!response.ok) {
      throw await ResponseError.fromResponse(response);
    }

    return asProjectUser(await response.json());
  }

  async delete(
    deliveryProjectUserId: string,
    deliveryProjectId: string,
  ): Promise<void> {
    const url = await this.#getBaseUrl();

    const body: DeleteDeliveryProjectUserRequest = {
      delivery_project_user_id: deliveryProjectUserId,
      delivery_project_id: deliveryProjectId,
    };

    const response = await this.#fetchApi.fetch(url, {
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(body),
    });

    if (!response.ok) {
      throw await ResponseError.fromResponse(response);
    }
  }
}

function asProjectUsers(result: DeliveryProjectUser[]) {
  return result.map(asProjectUser);
}

function asProjectUser(result: DeliveryProjectUser) {
  result.updated_at = new Date(result.updated_at);
  return result;
}
