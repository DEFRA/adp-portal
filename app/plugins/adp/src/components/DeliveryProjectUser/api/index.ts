export {
  deliveryProjectUserApiRef,
  type DeliveryProjectUserApi,
} from './DeliveryProjectUserApi';
export { DeliveryProjectUserClient } from './DeliveryProjectUserClient';
