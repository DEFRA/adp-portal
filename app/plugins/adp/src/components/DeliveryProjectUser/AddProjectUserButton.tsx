import { Button } from '@material-ui/core';
import React, { useState } from 'react';
import { deliveryProjectUserApiRef } from './api';
import { alertApiRef, useApi } from '@backstage/core-plugin-api';
import {
  DeliveryProjectUserFormFields,
  emptyForm,
  type DeliveryProjectUserFields,
} from './DeliveryProjectUserFormFields';
import { DialogForm, TitleWithHelp, readValidationError } from '../../utils';
import type { SubmitResult } from '../../utils';
import { usePermission } from '@backstage/plugin-permission-react';
import { deliveryProjectUserCreatePermission } from '@internal/plugin-adp-common';

export type AddProjectUserButtonProps = Readonly<
  Omit<Parameters<typeof Button>[0], 'onClick'> & {
    onCreated?: () => void;
    deliveryProjectId: string;
    entityRef: string;
  }
>;

export function AddProjectUserButton({
  onCreated,
  deliveryProjectId,
  entityRef,
  children,
  ...buttonProps
}: AddProjectUserButtonProps) {
  const [isModalOpen, setIsModalOpen] = useState(false);
  const alertApi = useApi(alertApiRef);
  const client = useApi(deliveryProjectUserApiRef);
  const { allowed: canCreateProjectUser } = usePermission({
    permission: deliveryProjectUserCreatePermission,
    resourceRef: deliveryProjectId,
  });

  if (!canCreateProjectUser) return null;

  async function handleSubmit(
    fields: DeliveryProjectUserFields,
  ): Promise<SubmitResult<DeliveryProjectUserFields>> {
    try {
      await client.create({
        delivery_project_id: deliveryProjectId,
        ...fields,
      });
    } catch (e: any) {
      return readValidationError(e);
    }
    alertApi.post({
      message: 'Delivery Project User added successfully',
      severity: 'success',
      display: 'transient',
    });
    return { type: 'success' };
  }

  return (
    <>
      <Button
        {...buttonProps}
        onClick={() => setIsModalOpen(true)}
        data-testid="add-project-user-button"
      >
        {children}
      </Button>
      {isModalOpen && (
        <DialogForm
          defaultValues={emptyForm}
          renderFields={DeliveryProjectUserFormFields}
          completed={success => {
            setIsModalOpen(false);
            if (success) onCreated?.();
          }}
          title={
            <TitleWithHelp href="https://defra.github.io/adp-documentation/Getting-Started/onboarding-a-user/">
              Add team member
            </TitleWithHelp>
          }
          confirm="Add"
          submit={handleSubmit}
        />
      )}
    </>
  );
}
