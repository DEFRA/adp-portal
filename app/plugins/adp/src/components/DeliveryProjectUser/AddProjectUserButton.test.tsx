import React from 'react';
import { alertApiRef, type AlertApi } from '@backstage/core-plugin-api';
import { deliveryProjectUserApiRef, type DeliveryProjectUserApi } from './api';
import { act, render, waitFor } from '@testing-library/react';
import type { AddProjectUserButtonProps } from './AddProjectUserButton';
import { AddProjectUserButton } from './AddProjectUserButton';
import { TestApiProvider } from '@backstage/test-utils';
import {
  DeliveryProjectUserFormFields,
  emptyForm,
  type DeliveryProjectUserFields,
} from './DeliveryProjectUserFormFields';
import type * as DialogFormModule from '../../utils/DialogForm';
import userEvent from '@testing-library/user-event';
import type { ValidationError as IValidationError } from '@internal/plugin-adp-common';
import { ValidationError } from '../../utils';

function setup() {
  const mockAlertApi: jest.Mocked<AlertApi> = {
    alert$: jest.fn(),
    post: jest.fn(),
  };
  const mockProjectUserApi: jest.Mocked<DeliveryProjectUserApi> = {
    create: jest.fn(),
    getAll: jest.fn(),
    getByDeliveryProjectId: jest.fn(),
  };

  return {
    mockAlertApi,
    mockProjectUserApi,
    async renderComponent(props: AddProjectUserButtonProps) {
      const result = render(
        <TestApiProvider
          apis={[
            [alertApiRef, mockAlertApi],
            [deliveryProjectUserApiRef, mockProjectUserApi],
          ]}
        >
          <AddProjectUserButton {...props} />
        </TestApiProvider>,
      );
      await waitFor(() => expect(result.baseElement).not.toBeEmptyDOMElement());
      return { result };
    },
  };
}

const fields: DeliveryProjectUserFields = {
  user_catalog_name: 'user-1234',
  github_username: 'user-1234',
  is_admin: false,
  is_technical: true,
};

const DialogForm: jest.MockedFn<typeof DialogFormModule.DialogForm> = jest.fn();

jest.mock(
  '../../utils/DialogForm',
  () =>
    ({
      get DialogForm() {
        return DialogForm as typeof DialogFormModule.DialogForm;
      },
    } satisfies typeof DialogFormModule),
);

describe('AddProjectUserButton', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should only render a button initially', async () => {
    const { renderComponent, mockAlertApi, mockProjectUserApi } = setup();

    const { result } = await renderComponent({
      content: 'Test button',
      deliveryProjectId: '123',
    });

    expect(result.baseElement).toMatchSnapshot();
    expect(mockAlertApi.alert$).not.toHaveBeenCalled();
    expect(mockAlertApi.post).not.toHaveBeenCalled();
    expect(mockProjectUserApi.create).not.toHaveBeenCalled();
    expect(mockProjectUserApi.getAll).not.toHaveBeenCalled();
    expect(mockProjectUserApi.getByDeliveryProjectId).not.toHaveBeenCalled();
  });

  it('should render the dialog when the button is clicked', async () => {
    const { renderComponent, mockAlertApi, mockProjectUserApi } = setup();
    DialogForm.mockReturnValue(<span>Test dialog</span>);

    const { result } = await renderComponent({
      content: 'Test button',
      deliveryProjectId: '123',
    });
    await userEvent.click(result.getByTestId('add-project-user-button'));

    expect(result.baseElement).toMatchSnapshot();
    expect(DialogForm.mock.calls).toHaveLength(1);
    const formProps = DialogForm.mock.calls[0][0];
    expect({
      open: formProps.open,
      renderFields: formProps.renderFields,
      confirm: formProps.confirm,
      cancel: formProps.cancel,
      defaultValues: formProps.defaultValues,
      disabled: formProps.disabled,
      validate: formProps.validate,
    }).toMatchObject({
      open: undefined,
      renderFields: DeliveryProjectUserFormFields,
      confirm: 'Add',
      cancel: undefined,
      defaultValues: emptyForm,
      disabled: undefined,
      validate: undefined,
    });
    expect(formProps.submit).toBeDefined();
    expect(formProps.completed).toBeDefined();
    expect(mockAlertApi.alert$).not.toHaveBeenCalled();
    expect(mockAlertApi.post).not.toHaveBeenCalled();
    expect(mockProjectUserApi.create).not.toHaveBeenCalled();
    expect(mockProjectUserApi.getAll).not.toHaveBeenCalled();
    expect(mockProjectUserApi.getByDeliveryProjectId).not.toHaveBeenCalled();
  });

  it('should close the dialog when the form is cancelled', async () => {
    const { renderComponent, mockAlertApi, mockProjectUserApi } = setup();
    DialogForm.mockReturnValue(<span>Test dialog</span>);

    const { result } = await renderComponent({
      content: 'Test button',
      deliveryProjectId: '123',
    });
    await userEvent.click(result.getByTestId('add-project-user-button'));

    expect(result.baseElement).toMatchSnapshot('Before cancel');
    expect(DialogForm.mock.calls).toHaveLength(1);
    const formProps = DialogForm.mock.calls[0][0];
    act(() => formProps.completed(undefined));
    await waitFor(() => expect(result.queryByText('Test dialog')).toBeNull());
    expect(result.baseElement).toMatchSnapshot('After cancel');
    expect(mockAlertApi.alert$).not.toHaveBeenCalled();
    expect(mockAlertApi.post).not.toHaveBeenCalled();
    expect(mockProjectUserApi.create).not.toHaveBeenCalled();
    expect(mockProjectUserApi.getAll).not.toHaveBeenCalled();
    expect(mockProjectUserApi.getByDeliveryProjectId).not.toHaveBeenCalled();
  });

  it('should call onCreated when the form closes with a value', async () => {
    const { renderComponent, mockAlertApi, mockProjectUserApi } = setup();
    DialogForm.mockReturnValue(<span>Test dialog</span>);
    const onCreated: jest.MockedFn<
      Exclude<AddProjectUserButtonProps['onCreated'], undefined>
    > = jest.fn();

    const { result } = await renderComponent({
      content: 'Test button',
      deliveryProjectId: '123',
      onCreated,
    });
    await userEvent.click(result.getByTestId('add-project-user-button'));

    expect(result.baseElement).toMatchSnapshot('Before complete');
    expect(DialogForm.mock.calls).toHaveLength(1);
    expect(onCreated).not.toHaveBeenCalled();
    const formProps = DialogForm.mock.calls[0][0];
    act(() => formProps.completed(fields));
    await waitFor(() => expect(result.queryByText('Test dialog')).toBeNull());
    expect(result.baseElement).toMatchSnapshot('After complete');
    expect(onCreated).toHaveBeenCalledTimes(1);
    expect(mockAlertApi.alert$).not.toHaveBeenCalled();
    expect(mockAlertApi.post).not.toHaveBeenCalled();
    expect(mockProjectUserApi.create).not.toHaveBeenCalled();
    expect(mockProjectUserApi.getAll).not.toHaveBeenCalled();
    expect(mockProjectUserApi.getByDeliveryProjectId).not.toHaveBeenCalled();
  });

  it('should call the API when submitting', async () => {
    const { renderComponent, mockAlertApi, mockProjectUserApi } = setup();
    DialogForm.mockReturnValue(<span>Test dialog</span>);

    const { result } = await renderComponent({
      content: 'Test button',
      deliveryProjectId: 'project-1',
    });
    await userEvent.click(result.getByTestId('add-project-user-button'));

    expect(result.baseElement).toMatchSnapshot();
    expect(DialogForm.mock.calls).toHaveLength(1);
    const formProps = DialogForm.mock.calls[0][0];
    expect(mockAlertApi.alert$).not.toHaveBeenCalled();
    expect(mockAlertApi.post).not.toHaveBeenCalled();
    expect(mockProjectUserApi.create).not.toHaveBeenCalled();
    expect(mockProjectUserApi.getAll).not.toHaveBeenCalled();
    expect(mockProjectUserApi.getByDeliveryProjectId).not.toHaveBeenCalled();

    const submitResult = await formProps.submit(fields);
    expect(submitResult).toMatchObject({ type: 'success' });
    expect(mockProjectUserApi.create.mock.calls).toMatchObject([
      [
        {
          delivery_project_id: 'project-1',
          github_username: 'user-1234',
          is_admin: false,
          is_technical: true,
          user_catalog_name: 'user-1234',
        },
      ],
    ]);
    expect(mockProjectUserApi.getAll).not.toHaveBeenCalled();
    expect(mockProjectUserApi.getByDeliveryProjectId).not.toHaveBeenCalled();
    expect(mockAlertApi.alert$).not.toHaveBeenCalled();
    expect(mockAlertApi.post.mock.calls).toMatchObject([
      [
        {
          message: 'Delivery Project User added successfully',
          severity: 'success',
          display: 'transient',
        },
      ],
    ]);
  });

  it('should catch validation errors when submitting', async () => {
    const { renderComponent, mockAlertApi, mockProjectUserApi } = setup();
    DialogForm.mockReturnValue(<span>Test dialog</span>);
    const validationErrors: IValidationError[] = [
      {
        path: 'abc',
        error: {
          message: 'Something broke',
        },
      },
    ];
    mockProjectUserApi.create.mockRejectedValueOnce(
      new ValidationError(validationErrors),
    );

    const { result } = await renderComponent({
      content: 'Test button',
      deliveryProjectId: 'project-2',
    });
    await userEvent.click(result.getByTestId('add-project-user-button'));

    expect(result.baseElement).toMatchSnapshot();
    expect(DialogForm.mock.calls).toHaveLength(1);
    const formProps = DialogForm.mock.calls[0][0];
    expect(mockProjectUserApi.create).not.toHaveBeenCalled();
    expect(mockProjectUserApi.getAll).not.toHaveBeenCalled();
    expect(mockProjectUserApi.getByDeliveryProjectId).not.toHaveBeenCalled();

    const submitResult = await formProps.submit(fields);
    expect(submitResult).toMatchObject({
      type: 'validationError',
      errors: validationErrors,
    });
    expect(mockProjectUserApi.create.mock.calls).toMatchObject([
      [
        {
          delivery_project_id: 'project-2',
          github_username: 'user-1234',
          is_admin: false,
          is_technical: true,
          user_catalog_name: 'user-1234',
        },
      ],
    ]);
    expect(mockProjectUserApi.getAll).not.toHaveBeenCalled();
    expect(mockProjectUserApi.getByDeliveryProjectId).not.toHaveBeenCalled();
    expect(mockAlertApi.alert$).not.toHaveBeenCalled();
    expect(mockAlertApi.post).not.toHaveBeenCalled();
  });
});
