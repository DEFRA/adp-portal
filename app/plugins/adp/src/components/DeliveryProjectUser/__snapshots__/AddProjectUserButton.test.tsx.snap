// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddProjectUserButton should call onCreated when the form closes with a value: After complete 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      content="Test button"
      data-testid="add-project-user-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      />
      <span
        class="MuiTouchRipple-root"
      >
        <span
          class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
          style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
          />
        </span>
      </span>
    </button>
  </div>
</body>
`;

exports[`AddProjectUserButton should call onCreated when the form closes with a value: Before complete 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      content="Test button"
      data-testid="add-project-user-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      />
      <span
        class="MuiTouchRipple-root"
      >
        <span
          class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
          style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
          />
        </span>
      </span>
    </button>
    <span>
      Test dialog
    </span>
  </div>
</body>
`;

exports[`AddProjectUserButton should call the API when submitting 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      content="Test button"
      data-testid="add-project-user-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      />
      <span
        class="MuiTouchRipple-root"
      >
        <span
          class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
          style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
          />
        </span>
      </span>
    </button>
    <span>
      Test dialog
    </span>
  </div>
</body>
`;

exports[`AddProjectUserButton should catch validation errors when submitting 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      content="Test button"
      data-testid="add-project-user-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      />
      <span
        class="MuiTouchRipple-root"
      >
        <span
          class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
          style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
          />
        </span>
      </span>
    </button>
    <span>
      Test dialog
    </span>
  </div>
</body>
`;

exports[`AddProjectUserButton should close the dialog when the form is cancelled: After cancel 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      content="Test button"
      data-testid="add-project-user-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      />
      <span
        class="MuiTouchRipple-root"
      >
        <span
          class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
          style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
          />
        </span>
      </span>
    </button>
  </div>
</body>
`;

exports[`AddProjectUserButton should close the dialog when the form is cancelled: Before cancel 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      content="Test button"
      data-testid="add-project-user-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      />
      <span
        class="MuiTouchRipple-root"
      >
        <span
          class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
          style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
          />
        </span>
      </span>
    </button>
    <span>
      Test dialog
    </span>
  </div>
</body>
`;

exports[`AddProjectUserButton should only render a button initially 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      content="Test button"
      data-testid="add-project-user-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      />
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;

exports[`AddProjectUserButton should render the dialog when the button is clicked 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      content="Test button"
      data-testid="add-project-user-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      />
      <span
        class="MuiTouchRipple-root"
      >
        <span
          class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
          style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
          />
        </span>
      </span>
    </button>
    <span>
      Test dialog
    </span>
  </div>
</body>
`;
